src/config.js

```diff
@@ -0,0 +1,391 @@
+    assert(typeof addressType === 'string', "Configuration file needs addressType.")
+    this.addressType = addressType;
+
+    assert(
+      extendedPublicKeys && 
+      Array.isArray(extendedPublicKeys),
+      "Configuration file needs array of extendedPublicKeys."
+    ) 
+    this.extendedPublicKeys = extendedPublicKeys;
+    this.validateExtendedPublicKeys(false);
+
+    if (startingAddressIndex) {
+      if (typeof startingAddressIndex !== 'number') {
+        throw new TypeError('Expected int for startingAddressIndex');
+      }
+      this.startingAddressIndex = startingAddressIndex;
+    }
+  }
+
+  /**
+   * @description instantiate a new wallet config object from a JSON
+   * @param {string} jsonString - json string to convert.
+   * @returns {MultisigWalletConfig} returns a new instance of a MultisigWalletConfig
+   */
+  static fromJSON(jsonString) {
+    if (typeof jsonString !== 'string') throw new TypeError('Must pass a valid JSON string');
+    let options;
+
+    try {
+      options = JSON.parse(jsonString);
+    } catch (error) {
+      throw new Error("Unable to parse JSON.");
+    }
+    
+    // TODO: how do we handle caravan configs where the extended public keys
+    // don't include xfp information? We can get a parent fingerprint 
+    // from the xpub and use that. It wouldn't be valid as an xfp but could be used as stand in? 
+    return new this(options);
+  }
+
+  /**
+   * @description static method wrapper around getting a new
+   * config based on a JSON since Caravan is the assumed format of accepted JSON
+   * @param {string} jsonString - valid config json string from Caravan export
+   * @returns {MultisigWalletConfig} new instance of a MultisigWalletConfig
+   */
+  static fromCaravanConfig(jsonString) {
+    return MultisigWalletConfig.fromJSON(jsonString);
+  }
+
+  /**
+   * @description Takes a coldcard config string and creates
+   * a new instance of a MultisigWalletConfig.
+   * 
+   * NOTE for ColdCard: only the root xfp of the signing device is required to be correct, but we
+   * recommend only setting up the multisig wallet on the Coldcard with complete xfp
+   * information.
+   * 
+   * This is an example Coldcard config file from
+   * https://coldcardwallet.com/docs/multisig
+   * 
+   * # Coldcard Multisig setup file (exported from 4369050F)
+   * #
+   * Name: MeMyself
+   * Policy: 2 of 4
+   * Derivation: m/45'
+   * Format: P2WSH
+   * 
+   * D0CFA66B: tpubD9429UXFGCTKJ9NdiNK4rC5...DdP9
+   * 8E697B74: tpubD97nVL37v5tWyMf9ofh5rzn...XgSc
+   * BE26B07B: tpubD9ArfXowvGHnuECKdGXVKDM...FxPa
+   * 4369050F: tpubD8NXmKsmWp3a3DXhbihAYbY...9C8n
+   * 
+   * @param {string} configString valid coldcard config string
+   * @returns {MultisigWalletConfig} new instance of MultisigWalletConfig
+   * @example
+   * import { MultisigWalletConfig } from "unchained-wallets";
+   * 
+   * const config = MultisigWalletConfig.fromColdCardConfig(ccFileData);
+   * console.log(JSON.parse(config.toJSON()))
```

Comment:
I think this is toJSONConfig ? And in this example - it doesnt return startingAddress or client because they're unknown, right?

Reply:
Yeah, that's right. They don't exist on the CC config so they will be skipped (and they're not required in the constructor). 

I'm now wondering if `toJSONConfig` is the right name for the method or if `toJSON` is better. Let me know what you think, but updating the comment to reflect the name of the method as it is now (as per your comment).